drivers/ethernet/eth_stm32_hal.c

```diff
@@ -270,41 +274,61 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 
 	heth = &dev_data->heth;
 
-	k_mutex_lock(&dev_data->tx_mutex, K_FOREVER);
-
 	total_len = net_pkt_get_len(pkt);
 	if (total_len > ETH_TX_BUF_SIZE) {
 		LOG_ERR("PKT too big");
-		res = -EIO;
-		goto error;
+		return -EIO;
+	}
+
+	k_mutex_lock(&dev_data->tx_mutex, K_FOREVER);
```

Comment:
Changes to existing code should be done in a dedcaited commit with justification. Ideally a dedicated PR if there are multiple.

Reply:
Here it is: 34898e22d8a7